docs/troubleshooting.md

```diff
@@ -36,15 +36,11 @@ An usual symptom is the presence of errors containing `unknown symbol` and it is
 
 The workaround is to use a version of haskell-language-server compiled from source with the ghc option `-dynamic` enabled. See more details [here](https://github.com/haskell/haskell-language-server/issues/1160#issuecomment-756566273).
 
-### Problems with Template Haskell
+### Support for Template Haskell
 
-Due to how Template Haskell code is evaluated at compile time and some limitations in the interaction between HLS and GHC, the loading of modules using TH can be problematic.
+Template Haskell should work fine in Linux and Windows with the distributed binaries. In Mac Os a dynamically linked binary of HLS is required to avoid segmentation faults. The easiest way to obtain a dynamically linked HLS binary is to build it locally. With cabal install this can be done as follows:
 
-The errors thrown are usually related to linking and usually make HLS crash: `Segmentation fault`, `GHC runtime linker: fatal error`, etc
-
-A workaround which has helped in some cases is to compile HLS from source with the ghc option `-dynamic` enabled, as in the previous issue.
-
-We have a [dedicated label](https://github.com/haskell/haskell-language-server/issues?q=is%3Aissue+is%3Aopen+label%3A%22type%3A+template+haskell+related%22) in the issue tracker and an [general issue](https://github.com/haskell/haskell-language-server/issues/1431) tracking support for TH.
+    cabal update && cabal install haskell-language-server --enable-executable-dynamic
```

Comment:
@jneira One issue is that the command line above doesn't work for GHC 9.0 as certain plugins (class, tactics, etc.) require flags to be disabled. This goes back to #2317 I'm afraid

Reply:
Nevermind, I see that #2322 should fix the issue